lib/routes/create_invoice/pay_invoice_page.dart

```diff
@@ -0,0 +1,500 @@
+                            }
+
+                            if (message != null) {
+                              // In case error doesn't have a trailing full stop
+                              if (!message.endsWith('.')) {
+                                message += '.';
+                              }
+                              return Container(
+                                  padding: EdgeInsets.only(
+                                      top: 50.0, left: 30.0, right: 30.0),
+                                  child: Column(children: <Widget>[
+                                    Text(
+                                      message,
+                                      textAlign: TextAlign.center,
+                                      style: theme.warningStyle.copyWith(
+                                          color: Theme.of(context).errorColor),
+                                    ),
+                                  ]));
+                            } else {
+                              return SizedBox();
+                            }
+                          })
+                    ],
+                  ),
+                ),
+              ),
+            ),
+          );
+        },
+      ),
+    );
+  }
+
+  Widget _buildPayableBTC(AccountModel acc) {
+    return GestureDetector(
+      child: AutoSizeText(
+        "Pay between ${acc.currency.format(widget.lnurlPay.minAmount)} and ${acc.currency.format(widget.lnurlPay.maxAmount)}",
+        style: theme.textStyle,
+        maxLines: 1,
+        minFontSize: MinFontSize(context).minFontSize,
+      ),
+      onTap: () => _amountController.text = acc.currency.format(
+          widget.lnurlPay.minAmount,
+          includeDisplayName: false,
+          userInput: true),
+    );
+  }
+
+  void applyPayFetchResponse(PayFetchResponse response, AccountModel account) {
+    _payFetchResponse = response;
+    _descriptionController.text = response.comment;
+    _amountController.text = account.currency.format(response.maxAmount,
+        includeDisplayName: false,
+        userInput:
+            true); // FINDOUT Why doesn't this trigger didChangeDependencies?
+  }
+
+  Future _getInvoice(InvoiceBloc invoiceBloc, AccountBloc accountBloc,
+      LNUrlBloc lnurlBloc, AccountModel account) {
+    /*
+         5. LN WALLET makes a GET request using callback with the following query parameters:
+         amount (input) - user specified sum in MilliSatoshi
+         nonce - an optional parameter used to prevent server response caching
+         fromnodes - an optional parameter with value set to comma separated nodeIds if payer wishes a service to provide payment routes starting from specified LN nodeIds
+         comment (input) - an optional parameter to pass the LN WALLET user's comment to LN SERVICE
+
+       */
+
+    _payFetchResponse.amount = account.currency.parse(_amountController.text);
+    lnurlBloc.newPayRequestSink.add(_payFetchResponse);
```

Comment:
At this point we are requesting the invoice (step 5 in the spec).
An alternative way to create a new payment implementation ( `LNURLPayDialog` ) would be to wait for the result and merge into the current flow of paying an invoice ( `PaymentRequestDialog` ).
To achieve the pattern of waiting we can add an AsyncAction that adds executes the request inside the bloc and return a promise with the result.
What do you think?

Reply:
> At this point we are requesting the invoice (step 5 in the spec).
> An alternative way to create a new payment implementation ( `LNURLPayDialog` ) would be to wait for the result and merge into the current flow of paying an invoice ( `PaymentRequestDialog` ).
> To achieve the pattern of waiting we can add an AsyncAction that adds executes the request inside the bloc and return a promise with the result.

When I obtain the `invoice` does it make sense to add it to add it to  `decodeInvoiceSink`?

  widget.invoiceBloc.decodeInvoiceSink.add(widget.payinfo.invoice)

I was hoping that:
- the invoice is decoded and added to `invoiceBlock.readyInvoicesStream`.
- the decoded invoice and payment flow is handled by `received_invoice_notification.dart`.
- Navigation route pops back to lnurl-pay flow so we can show `successAction`.

Does this make sense?

> What do you think?

I'll implement your suggestion and test. Thanks.

Reply:
Sounds good. That is what was in my mind as well.